src/sonic-config-engine/minigraph.py

```diff
@@ -260,23 +260,16 @@ def parse_png(png, hname, dpg_ecmp_content = None):
         if child.tag == str(QName(ns, "Devices")):
             for device in child.findall(str(QName(ns, "Device"))):
                 (lo_prefix, lo_prefix_v6, mgmt_prefix, mgmt_prefix_v6, name, hwsku, d_type, deployment_id, cluster, d_subtype) = parse_device(device)
-                device_data = {}
-                if hwsku != None:
-                    device_data['hwsku'] = hwsku
+                # 'lo_addr', 'mgmt_addr', 'type', 'hwsku' are must fields in config db
```

Comment:
Why are they must fields?
And we should use "mandatory true" in yang models for must field.

Reply:
Thanks for comments, maybe can't say must fields. Seems for historical reason, code never check if 'lo_addr' and 'mgmt_addr' keys are exists or not. In minigraph, 'lo_addr' and 'mgmt_addr' sometime may not exists, but after deploy mg we generated config_db.json, code parse config_db.json will access  'lo_addr' and 'mgmt_addr' without confirm if keys are exists. 
Another fix may find all places referring 'lo_addr' and 'mgmt_addr', but they are many places and for safety I decided to change the behavior back.

The description "# 'lo_addr', ..." here seems a little bit confuse, I will remove this description.

Reply:
When 'lo_addr' and 'mgmt_addr' keys do not exists in minigraph, the value in config_db will be ‘None’.
How does consumer for this table handle this value?

Reply:
minigraph.py will first save 'lo_addr':None to DEVICE_NEIGHBOE_METADATA 
```
if asic_name is None:
        results['DEVICE_NEIGHBOR_METADATA'] = { key:devices[key] for key in devices if key.lower() != hostname.lower() }
    else:
        results['DEVICE_NEIGHBOR_METADATA'] = { key:devices[key] for key in devices if key in {device['name'] for device in neighbors.values()} }
```

Then results to json will treat None as "None" in final config_db.json, and in show cli python code, it will access the value in config_db.json, than is "lo_addr":"None". So the final show result is something like this:

admin@str-dx010-acs-5:~$ show interface neighbor exp
LocalPort    Neighbor    NeighborPort    NeighborLoopback    NeighborMgmt    NeighborType
-----------  ----------  --------------  ------------------  --------------  --------------
Ethernet112  ARISTA01T1  Ethernet1       None                10.64.247.220   LeafRouter
Ethernet116  ARISTA02T1  Ethernet1       None                10.64.247.221   LeafRouter
Ethernet120  ARISTA03T1  Ethernet1       None                10.64.247.222   LeafRouter
Ethernet124  ARISTA04T1  Ethernet1       None                10.64.247.223   LeafRouter